irohad/model/execution/impl/command_executor.cpp

```diff
@@ -261,6 +262,69 @@ namespace iroha {
       return true;
     }
 
+      // ----------------------------|SubtractAssetQuantity|-----------------------------
+
+      SubtractAssetQuantityExecutor::SubtractAssetQuantityExecutor() {
+        log_ = logger::log("SubtractAssetQuantityExecutor");
+      }
+
+      bool SubtractAssetQuantityExecutor::execute(const Command &command,
+                                             WsvQuery &queries,
+                                             WsvCommand &commands) {
+        auto subtract_asset_quantity = static_cast<const SubtractAssetQuantity &>(command);
+
+        auto asset = queries.getAsset(subtract_asset_quantity.asset_id);
+        if (not asset.has_value()) {
```

Comment:
Use implicit cast to bool instead: `if (not asset) ...` it can be done also with boost::optional
Same below

Reply:
It's like other code below
https://github.com/hyperledger/iroha/blob/develop/irohad/model/execution/impl/command_executor.cpp#L206

https://github.com/hyperledger/iroha/blob/develop/irohad/model/execution/impl/command_executor.cpp#L620

How about it?

Reply:
Well, it would be nice to refactor those as well. As you wish

Reply:
Ok but I want to separate PR